### DISCUSSION TITLE
Places and Visits - how are they meant to work

### DISCUSSION THREAD

**Discussion by nubs** [Freika/dawarich#298] [Q&A] [answered]
I'm loving dawarich and the beta feature of places and visits seems like it could be very powerful, but in its current state it leaves me feeling like I'm using it wrongly. Each day, I get a bunch of "Suggested places" which sometimes have good recommendations but the suggestions never seem to pull from places that I have previously tagged (for example, Home, Work).

In addition, I see duplicate places appearing in the list that appear to have the same name, latitude, and longitude rather than all being consolidated underneath one place.
![image](https://github.com/user-attachments/assets/99df852a-dca8-4452-9749-dad0df2c247c)

In my mental model, a place would be a unique location that I have visited at some point in time and all visits to that place would reference the same place. For instance, if I go to work 5 days a week, there should only be one "Work" place but there would be 5 visits there for the 5 days.

Finally, it would be nice to have visits with higher degrees of confidence automatically assigned to places without having to select them each day. Given that I regularly go to "Work", I would love for dawarich to assume that is where I am visiting  if I'm within some distance (say 10m or 50ft) of Work.

Question: Does my mental model match the planned features for dawarich? Is my experience a bug, expected experience given the beta nature of the feature, or something else?

**Comment by Freika** [comment, marked as answer]
Visits and Places in their current form are very raw things, so they are far from perfect. In the ideal scenario, they will work pretty much as you have described. So, with that said, I'll be working on them to make them useful and usable. As for now, I'll probably disable visit suggestion background job so it wouldn't cause more chaos than it already did.

Follow the changelogs and one day you'll find an update to this feature :)

**Reply by nubs** [reply]
Thanks, that's what I was hoping. When the feature gets finalized it would be great to have a way of merging places as well!
